qiskit_machine_learning/neural_networks/neural_network.py

```diff
@@ -164,7 +164,8 @@ def _preprocess_forward(
                 num_samples = 1
                 parameters = np.broadcast_to(weights, (num_samples, len(weights)))
             else:
-                return None, None
+                # no input, no weights, just execute circuit once
+                return np.asarray([]), 1
```

Comment:
Pretty minor comment: I guess having a return was explicit here before, but this is just one case out of the 4 of the if/else logic, where the others set things up to use the return at the end. Maybe it would be nice to do the same here and set num_samples = 1 and parameters = np.asarray)[]) and use the common return at the end (in this case num_samples = 1 applies to the whole else here i.e. where input_data is None)

Reply:
It makes sense, updated.